all-your-base: Compilation warning on GCC 11.1.0

Compiling all-your-base with GCC 11.1.0 results in the following warning message (which causes an error when used like we do here with -Werror)

```
Running tests for all_your_base
Compiling memcheck
In file included from ./test-framework/unity.h:21,
                 from ./test_all_your_base.c:1:
./test_all_your_base.c: In function ‘test_empty_list’:
./test_all_your_base.c:109:32: error: ‘rebase’ accessing 64 bytes in a region of size 1 [-Werror=stringop-overflow=]
  109 |    TEST_ASSERT_EQUAL_size_t(0, rebase(input, 2, 10, 0));
./test-framework/unity_internals.h:785:156: note: in definition of macro ‘UNITY_TEST_ASSERT_EQUAL_UINT’
  785 | #define UNITY_TEST_ASSERT_EQUAL_UINT(expected, actual, line, message)                            UnityAssertEqualNumber((UNITY_INT)(expected), (UNITY_INT)(actual), (message), (UNITY_LINE_TYPE)(line), UNITY_DISPLAY_STYLE_UINT)
      |                                                                                                                                                            ^~~~~~
./test_all_your_base.c:109:4: note: in expansion of macro ‘TEST_ASSERT_EQUAL_size_t’
  109 |    TEST_ASSERT_EQUAL_size_t(0, rebase(input, 2, 10, 0));
      |    ^~~~~~~~~~~~~~~~~~~~~~~~
./test_all_your_base.c:109:32: note: referencing argument 1 of type ‘int8_t *’ {aka ‘signed char *’}
  109 |    TEST_ASSERT_EQUAL_size_t(0, rebase(input, 2, 10, 0));
./test-framework/unity_internals.h:785:156: note: in definition of macro ‘UNITY_TEST_ASSERT_EQUAL_UINT’
  785 | #define UNITY_TEST_ASSERT_EQUAL_UINT(expected, actual, line, message)                            UnityAssertEqualNumber((UNITY_INT)(expected), (UNITY_INT)(actual), (message), (UNITY_LINE_TYPE)(line), UNITY_DISPLAY_STYLE_UINT)
      |                                                                                                                                                            ^~~~~~
./test_all_your_base.c:109:4: note: in expansion of macro ‘TEST_ASSERT_EQUAL_size_t’
  109 |    TEST_ASSERT_EQUAL_size_t(0, rebase(input, 2, 10, 0));
      |    ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from ./test_all_your_base.c:4:
./all_your_base.h:9:8: note: in a call to function ‘rebase’
    9 | size_t rebase(int8_t digits[DIGITS_ARRAY_SIZE], int16_t from_base,
      |        ^~~~~~
cc1: all warnings being treated as errors
make: *** [makefile:27: memcheck] Error 1
```

--- a/exercises/practice/all-your-base/test_all_your_base.c
+++ b/exercises/practice/all-your-base/test_all_your_base.c
@@ -105,7 +105,7 @@ static void test_15_bit_integer(void)
 static void test_empty_list(void)
 {
    TEST_IGNORE();
-   int8_t input[] = { 0 };
+   int8_t input[DIGITS_ARRAY_SIZE] = { 0 };
    TEST_ASSERT_EQUAL_size_t(0, rebase(input, 2, 10, 0));
 }
 

